sm/plugin_sm/src/plugin_manager.rs

```diff
@@ -68,9 +76,34 @@ impl ExternalPlugins {
         for maybe_entry in fs::read_dir(&self.plugin_dir)? {
             let entry = maybe_entry?;
             let path = entry.path();
+            dbg!(&path);
             if path.is_file() {
-                // TODO check the file is exec
-                // TODO check the command is actually a plugin
+                match Command::new(&path).arg("list").status() {
+                    Ok(code) if code.success() => {}
+
+                    // If the file is not executable or returned non 0 status code we assume it is not a valid and skip further processing.
+                    Ok(_) => {
+                        info!("File {} in plugin directory does not support list operation and may not be a valid plugin, skipping.", pathbuf_to_string(path.clone()).unwrap());
```

Comment:
It's actually an error, so I suggest to use `error!` instead of `info!`. Also below.